src/interfaces/NamingService.ts

```diff
@@ -0,0 +1,15 @@
+import {  ResolutionMethod } from '../types/publicTypes';
+
+export default interface NamingService {
```

Comment:
I am not a big fan of that change. See, base abstract class allows for better code sharing. I never saw any project that would pursue pure interfaces internally. Interfaces are only for for external exports.
Internally, it is always an abstract class with base structure but never a pure interface.
You may see that accross libraries like web3 or ethers where they do have an interface but use abstract base class internally anyway. 

So now, as you sharing code between services through utilities it is harder to use. Like:

```
ensureConfigured({
  url: this.url, 
  network: source.network, 
  provider: this.provider
  this.name
);
```

Instead of:

```
this.ensureConfigured();
```

Reply:
@bogdan There were a lot of unnecessary dependencies between classes. Relatively small update of constructor signature caused changes across the whole library which i think is not right. Naming services had 4 levels of inheritance on a worst-case scenario and code wasn't looked readable for me.

As i can see - there wasn't much code sharing between naming services - but a lot changes, even internal, required updates across the all services. Now we share a few methods between naming services, i don't think it's enough to introduce inheritance.

I agree about methods behaviour inconsistency. Methods signature is covered by interfaces, but we need to provide consistent behavour in case of exception which could be done on Resolution service. What do you think?

Reply:
It is fine to remove shared constructor, but I think abstract base class with protected shared methods instead of an interface with utilities import is better.

Reply:
After finishing with these changes I felt the codebase become much easier. Usually, it takes me a while to jump between NamingService, EthereiumNamingService, and specific NamingService. It was mostly in the constructor though. ENS and CNS despite the fact they have many in common, I found them quite different in their implementation.

Also as I understand in the future we are going to add a bunch more naming services. I could group CNS and other UD-like naming services under 1 base class that implements NamingService interface without being afraid this will somehow interfere with ENS. This way they will get the shared functionality without duplicating too much of a code and it should still be much easier to read.